src/PdfReader/index.tsx

```diff
@@ -174,6 +211,41 @@ export default function usePdfReader(args: ReaderArguments): ReaderReturn {
     setPdfResource(manifest, proxyUrl);
   }, [proxyUrl, manifest]);
 
+  /**
+   * calculate the height or width of the pdf page in paginated mode.
+   *  - if the page's aspect ratio is taller than the container's, we will constrain
+   *    the page to the height of the container.
+   *  - if the page's aspect ratio is wider than the container's, we will constrain
+   *    the page to the width of the container
+   */
+
+  const resizePage = React.useCallback(
+    (
+      pdfWidth: number,
+      pdfHeight: number,
+      containerSize: { width: number; height: number }
+    ) => {
+      const wRatio = pdfWidth / containerSize.width;
+      const hRatio = pdfHeight / containerSize.height;
+
+      const fitHorizontal = wRatio > hRatio;
+      const width = fitHorizontal ? Math.round(containerSize.width) : undefined;
+      const height = !fitHorizontal
+        ? Math.round(containerSize.height)
+        : undefined;
+
+      dispatch({ type: 'RESIZE_PAGE', width, height });
+    },
+    []
+  );
+
+  //TODO: Somehow, this window size updates when height
+  React.useEffect(() => {
+    if (containerSize) {
+      resizePage(state.pdfWidth, state.pdfHeight, containerSize);
+    }
+  }, [containerSize, state.pdfWidth, state.pdfHeight, resizePage]);
```

Comment:
I don't think you need to define the `resizePage` outside the `useEffect`, I think it can just be written inline there, but I may be wrong?

Reply:
Nvm I see it is also used below